src/develop/imageop.c

```diff
@@ -1360,10 +1362,26 @@ void dt_iop_set_module_in_trouble(dt_iop_module_t *module, const gboolean state)
 static void _set_trouble_message(dt_iop_module_t *const module, const char* const trouble_msg,
                                  const char* const trouble_tooltip, const char *const stderr_message)
 {
-  GtkWidget *label_widget = module ? module->warning_label : NULL;
-  //TODO: write function to create the label widget on the module's header
-  //if (!label_widget)
-  //  label_widget = module->warning_label = create_warning_label(module);
+  GtkWidget *label_widget = NULL;
+
+  if(module && module->widget)
+  {
+    GtkWidget *iopw = gtk_widget_get_parent(module->widget);
```

Comment:
You should not unconditionally create the widget -- set_module_trouble_message gets called with NULL message every time the module's process() function is invoked to clear any prior error condition (unless there is in fact [still] an error to report).  So on the vast majority of calls, your code will be creating and then immediately destroying the widget.

Reply:
Makes a lot of sense! I think this is now fixed.